### PR TITLE
overlay: set permissions on "diff" correctly

### PR DESCRIPTION
**Pull request by nalind**
When creating a new layer that has a parent, assign the new layer's "diff" directory the same permissions as the parent layer's "diff" directory, rather than the permissions of the parent of its parent layer's "diff" directory.

Before the recent round of changes, that directory would be initialized with 0700 permissions, so layers created by new code on top of layers created by the older version would be marked 0700, and were not usable by UIDs other than 0.

Spotted in CI tests for buildah that try to use overlay to create containers on top of base images that were imported using a slightly older version of podman, for example https://cirrus-ci.com/task/6024517489262592.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @saschagrunert @TomSweeneyRedHat PTAL

**Review by vrothberg — COMMENTED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
LGTM

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -631,6 +626,18 @@ func (d *Driver) create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		}
 	}
 
+	perms := defaultPerms
+	if d.options.forceMask != nil {
```
Comment (giuseppe) [inline drivers/overlay/overlay.go:630]: I think this should have precedence and override the parent mode

Comment (nalind) [reply drivers/overlay/overlay.go:630]: Hmm, yeah, you're right.  Changing it.

**Pull request merged**